src/multi_vector_simulator/D2_model_constraints.py

```diff
@@ -75,35 +75,25 @@ def add_constraints(local_energy_system, dict_values, dict_model):
     - D2.test_add_constraints_net_zero_energy_requirement_is_true()
     - D2.test_add_constraints_net_zero_energy_requirement_is_false()
     """
-    count_added_constraints = 0
 
-    if dict_values[CONSTRAINTS][MINIMAL_RENEWABLE_FACTOR][VALUE] > 0:
-        # Add minimal renewable factor constraint
-        local_energy_system = constraint_minimal_renewable_share(
-            local_energy_system, dict_values, dict_model
-        )
-        count_added_constraints += 1
+    constraint_functions = {
+        MINIMAL_RENEWABLE_FACTOR: constraint_minimal_renewable_share,
+        MAXIMUM_EMISSIONS: constraint_maximum_emissions,
+        MINIMAL_DEGREE_OF_AUTONOMY: constraint_minimal_degree_of_autonomy,
+        NET_ZERO_ENERGY: constraint_net_zero_energy,
+    }
 
-    if dict_values[CONSTRAINTS][MAXIMUM_EMISSIONS][VALUE] is not None:
-        # Add maximum emissions constraint
-        local_energy_system = constraint_maximum_emissions(
-            local_energy_system, dict_values
-        )
-        count_added_constraints += 1
+    count_added_constraints = 0
 
-    if dict_values[CONSTRAINTS][MINIMAL_DEGREE_OF_AUTONOMY][VALUE] > 0:
-        # Add minimal renewable factor constraint
-        local_energy_system = constraint_minimal_degree_of_autonomy(
+    for constraint in dict_values[CONSTRAINTS]:
+        # if the constraint is not within its proper range of admissible values, None is returned
+        les = constraint_functions[constraint](
             local_energy_system, dict_values, dict_model
         )
-        count_added_constraints += 1
 
-    if dict_values[CONSTRAINTS][NET_ZERO_ENERGY][VALUE] == True:
-        # Add net zero energy (NZE) constraint
-        local_energy_system = constraint_net_zero_energy(
-            local_energy_system, dict_values, dict_model
-        )
-        count_added_constraints += 1
+        if les is not None:
+            local_energy_system = les
+            count_added_constraints += 1
```

Comment:
I think this way to write it is a bit confusing. You need to at least add a  comment explaining the `if les is not None`.

Why not use `local_energy_system` and `count_added_constraints` as in as well as output and increase `count_added_constraints` within the `constraint_functions`

Reply:
If the constraint fails is None or 0, it would return None and set `local_energy_system` to None, making the next loop possibly problematic

Reply:
I am not sure what you mean with 

> If the constraint fails is None or 0

You mean if the function of the constraint fails (ie. python error), or...?

Reply:
I mean if the MINIMUM_RENEWABLE_FACTOR is 0, then the content of the constraint function is skipped and `None` is returned instead of a instance of `local_energy_system`

Reply:
Then only a in-code comment has to be added.

Reply:
Done in 9f1300ac0ec7c37c5d87d4319be5fb2a1e014584